collator/src/mempool/impls/std_impl.rs

```diff
@@ -334,3 +340,198 @@ impl MempoolAdapter for MempoolAdapterStdImpl {
         Ok(())
     }
 }
+
+pub struct MempoolAdapterSingleNodeImpl(Arc<Inner>);
+
+struct Inner {
+    cache: Arc<Cache>,
+    config: Mutex<ConfigAdapter>,
+    node_peer_info: PeerInfo,
+
+    unprocessed_message_queue: Arc<std::sync::Mutex<VecDeque<Bytes>>>,
+    anchor_id: tokio::sync::Mutex<Cell<MempoolAnchorId>>,
+
+    once_flag: std::sync::Once,
+}
+
+impl Inner {
+    pub async fn start_process(&self) {
+        let consensus_config = {
+            let config_guard = self.config.lock().await;
+            let consensus_config = config_guard
+                .builder
+                .get_consensus_config()
+                .expect("There is no consensus config.");
+            consensus_config.clone()
+        };
+
+        let timeout = (consensus_config.broadcast_retry_millis
+            * (consensus_config.broadcast_retry_attempts as u16)) as u64;
+
+        let mut interval = tokio::time::interval(std::time::Duration::from_millis(timeout));
+
+        loop {
```

Comment:
We usually add some lifetime logs for background loops to see when (or "if") they are stopped

Smth like this at the beginning of the method:
```rust
scopeguard::defer! {
    tracing::info!("mempool background task stopped"); // message will be printed when the future is dropped
}
```